main.js

```diff
@@ -781,10 +781,26 @@ function sendMessage(text, user, chatId, options) {
         if (typeof user !== 'string' && !(user instanceof Array)) {
             adapter.log.warn('Invalid type of user parameter: ' + typeof user + '. Expected is string or array.');
         }
+	    
+	if (typeof user === 'string') {
+            var user2;
```

Comment:
please use let and not var

Reply:
> Sorry, but we should fix blockly instead of adding such a bad hack ... This makes no sense that way sorry. Can you please open an issue with a blockyl example that shows blockly graphci and the generated code?

Sure i can do this, but.. isn't it exactly this, what the function sendMessage does? Determine on the input data what to do and call _sendMessageHelper?

Reply:
sendMessage first of all is an internal message and if there is a "user" parameter passed it is a user and nothing else :-)
I think blockly is using sendTo to send a message and then you end here https://github.com/iobroker-community-adapters/ioBroker.telegram/blob/725e8c3b85f8f5b708a3d36812d75d2318930b94/main.js#L984 ... but also the mesaage has clear defined fields. So if blockly generates code which is "crap" and sends wrong messages then the blockly.js needs to be fixed in https://github.com/iobroker-community-adapters/ioBroker.telegram/blob/master/admin/blockly.js#L352 or such

Reply:
The issue #214 is the exact same thing. I leave a reply there with additional informations.